### PR TITLE
Add telemetry

### PR DESCRIPTION
**Pull request by ankitkumarr**
First take at adding basic telemetry to the CLI. 
I will be performing tests to make sure we log most of the scenarios possible and that we don't log any PII.

To test, please add the Instrumentation Key for you App Insights app. 

UPDATE - 
Here's how the Telemetry information looks in a basic command prompt-
telemetryoptoutlook

UPDATE 2- 
Here's the other PRs for the build systems we have-
https://github.com/Azure/homebrew-functions/pull/15
https://github.com/ahmelsayed/publish-script/pull/1

### CODE REVIEW

File: src/Azure.Functions.Cli/Helpers/TelemetryHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            StringBuilder sb = new StringBuilder();
+            // Append the bytes in hexadecimal form to the string builder
+            hashedID.ToList().ForEach(bt => sb.Append(bt.ToString("x2")));
+            return sb.ToString();
+        }
+
+        public static void LogEventIfAllowed(ConsoleAppLogEvent consoleEvent)
+        {
+            // This should be injected during build
+            var instrumentationKey = Guid.Empty;
```
Comment (ahmelsayed) [inline src/Azure.Functions.Cli/Helpers/TelemetryHelpers.cs]: It'll be easier to inject the value if you set it to

```csharp
var instrumentationKey  = Guid.Parse("00000000-0000-0000-0000-000000000000");
```

Comment (ahmelsayed) [reply src/Azure.Functions.Cli/Helpers/TelemetryHelpers.cs]: or the actual string value can also come from a const in the Constants class

**Comment by ahmelsayed** [comment]
Couple of more things to do sorry :) 

Can you update the [npm install script](https://github.com/Azure/azure-functions-core-tools/blob/master/src/Azure.Functions.Cli/npm/lib/install.js#L66-L70) to display information about the opt-out? Check with @ahmedelnably about the exact wording. We can probably use similar wording to dotnet cli. 

and also update the build script by adding a new task before [`DotnetPublish`](https://github.com/Azure/azure-functions-core-tools/blob/master/build/Program.cs#L16) that does a string replace on that file with the right env var.

**Comment by ankitkumarr** [comment]
Here's how the Telemetry information looks in a basic command prompt-

telemetryoptoutlook

**Comment by ankitkumarr** [comment]
It might be nice to include the version of the core-tools with each telemetry entry.

**Comment by riverar** [comment]
Opt-in telemetry like this isn't GDPR compliant, beware.

**Comment by ankitkumarr** [comment]
Closing in favor of https://github.com/Azure/azure-functions-core-tools/pull/1532

**Pull request closed without merging**